### ISSUE TITLE
Players are invisible to others when killed and infected, respawning the same tick as death while disguised.

### ISSUE THREAD

**Issue by mibby** [Eredrim/pvparena#53]
PVPArena dev 341
Modules dev 213
Paper dev 101 (Spigot 1.16.1)
LibsDisguises dev 674

FREE players spawn with an item that opens a custom GUI that lets them disguise as a block from console via commands such as ``/disguiseplayer %player% falling_block MELON setGridLocked``. 

Example Disguise GUI
https://paste.ubuntu.com/p/Y4kmq7wYWs/

Arena Config - BlockHunt
https://paste.ubuntu.com/p/FYMh7TrnGm/

LibsDisguise Config
https://paste.ubuntu.com/p/nKD7mFSG7V/

When an INFECTED player kills a FREE player, they respawn on the infected team and undisguised, per the betterclasses module;

```
    respawnCommand:
      free: ''
      custom: ''
      '%infected%': undisguiseplayer %player%
```

However upon respawning / dying while disguised, they are **invisible** to other infected players. I assume this is caused by handling respawning and undisguising at the same tick/time. As when players are killed while _not_ disguised, they respawn being visible to others on the infected team perfectly fine.

Toggling the setting ``preventDeath: false`` and _removing_ the respawnCommand does _not_ fix the issue, as players are not undisguised when killed and sometimes end up in a death loop / visual glitch.

Edit: Had a death respawn bug / loop even with ``preventDeath: true``, so assume it's also caused by teleporting and undisguising same tick? :(

**Comment by Eredrim** [comment]
Should be fixed in PVPArena 2.0 thanks to new death processing.

**Comment by mibby** [comment]
Great, thanks! Hopefully it isn't too far away before ready for testing.

**Comment by Eredrim** [comment]
Fixed in 2.0 dev preview
